packages/video_player/video_player_windows/windows/video_player.h

```diff
@@ -0,0 +1,106 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#ifndef PACKAGES_VIDEO_PLAYER_VIDEO_PLAYER_WINDOWS_WINDOWS_VIDEO_PLAYER_H_
+#define PACKAGES_VIDEO_PLAYER_VIDEO_PLAYER_WINDOWS_WINDOWS_VIDEO_PLAYER_H_
+
+#include <flutter/event_channel.h>
+#include <flutter/event_stream_handler.h>
+#include <flutter/event_stream_handler_functions.h>
+#include <flutter/standard_method_codec.h>
+#include <windows.h>
+#undef GetCurrentTime
+#include <shobjidl.h>
+#include <unknwn.h>
+#include <wincodec.h>
+
+// STL headers.
+#include <functional>
+#include <future>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <sstream>
+#include <string>
+
+// Include prior to C++/WinRT Headers.
+#include <wil/cppwinrt.h>
+
+// Windows Implementation Library.
+#include <wil/resource.h>
+
+// MediaFoundation headers.
+#include <Audioclient.h>
+#include <d3d11.h>
+#include <mfapi.h>
+#include <mferror.h>
+#include <mfmediaengine.h>
+#include <wincodec.h>
+
+#include "MediaEngineWrapper.h"
+#include "MediaFoundationHelpers.h"
+#include "messages.h"
+
+namespace video_player_windows {
+
+class VideoPlayer {
+ public:
+  VideoPlayer(IDXGIAdapter* adapter, HWND window, std::wstring uri,
+              flutter::EncodableMap http_headers);
+
+  virtual ~VideoPlayer() = default;
```

Comment:
Should `~VideoPlayer` call `Dispose`? Alternatively, should the destructor verify that `Dispose` was called?

Reply:
The plugin's Dispose is now calling dispose for everything.